### PR TITLE
[KWSys] add new port

### LINKED ISSUE #10203: [New Port Request] KWSys
**Library name:**

KWSys

**Library description:**

KWSys is the Kitware System Library.  It provides platform-independent
APIs to many common system features that are implemented differently on
every platform.  This library is intended to be shared among many
projects at the source level, so it has a configurable namespace.

**Source repository URL:**

https://gitlab.kitware.com/utils/kwsys

### PR DESCRIPTION
**Pull request by q1000treadz**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #10203

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  All. I tested on  `x64-windows`.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes.

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/kwsys/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO utils/kwsys
+    REF  dbc94f37f9821899aad441bcab525ad96a3f30dc #2021-08-06
+    SHA512 98441866fd51f2d4d3974d3c37d4456ce9e50c6f6c2ab0691e55b268907611ef061562fc30b1baa42aa195caf2281aa0e1e0799abc545fe6dae70efe2050ea50
+    HEAD_REF master
+)
+
+vcpkg_configure_cmake(
+	SOURCE_PATH ${SOURCE_PATH}
+	PREFER_NINJA
```
Comment (JonLiu1993) [inline ports/kwsys/portfile.cmake:12]:
```suggestion
vcpkg_cmake_configure(
	SOURCE_PATH "${SOURCE_PATH}"
```

File: ports/kwsys/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+		-DKWSYS_INSTALL_INCLUDE_DIR=${CURRENT_PACKAGES_DIR}/include
+    OPTIONS_RELEASE
+		-DKWSYS_INSTALL_BIN_DIR=${CURRENT_PACKAGES_DIR}/bin
+		-DKWSYS_INSTALL_LIB_DIR=${CURRENT_PACKAGES_DIR}/lib
+    OPTIONS_DEBUG
+		-DKWSYS_INSTALL_BIN_DIR=${CURRENT_PACKAGES_DIR}/debug/bin
+		-DKWSYS_INSTALL_LIB_DIR=${CURRENT_PACKAGES_DIR}/debug/lib
+)
+
+vcpkg_install_cmake()
```
Comment (JonLiu1993) [inline ports/kwsys/portfile.cmake:23]:
```suggestion
vcpkg_cmake_install()
```

File: ports/kwsys/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "kwsys",
+  "version-date": "2021-08-06",
+  "port-version": 2,
```
Comment (JonLiu1993) [inline ports/kwsys/vcpkg.json:4]:
```suggestion
```

File: ports/kwsys/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,26 @@
+vcpkg_from_gitlab(
```
Comment (JonLiu1993) [inline ports/kwsys/portfile.cmake:1]:
```suggestion
vcpkg_fail_port_install(ON_TARGET "UWP")
vcpkg_from_gitlab(
```

File: ports/kwsys/vcpkg.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+  "description": "Provides platform-independent APIs to many common system features",
+  "homepage": "https://gitlab.kitware.com/utils/kwsys",
+  "supports": "!uwp"
+}
```
Comment (JonLiu1993) [inline ports/kwsys/vcpkg.json:8]:
```suggestion
  "homepage": "https://gitlab.kitware.com/utils/kwsys",
  "supports": "!uwp",
  "dependencise":[
    {
      "name": "vcpkg-cmake",
      "host": true
    },
    {
      "name": "vcpkg-cmake-config",
      "host": true
    }
  ]
}
```

File: versions/k-/kwsys.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+      "version-date": "2021-08-06",
+      "port-version": 1
+    },
+    {
+      "git-tree": "4af210c9dac21c8c469a1f201f207a3dc4d12ec4",
+      "version-date": "2021-08-06",
+      "port-version": 0
+    }
+  ]
+}
```
Comment (JonLiu1993) [inline versions/k-/kwsys.json:19]:
```suggestion
{
  "versions": [
    {
      "git-tree": "3fb2fdfc0e8ecaa32c7173365698c73827add230",
      "version-date": "2021-08-06",
      "port-version": 0
    }
  ]
}
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/kwsys/vcpkg.json [hunk 7]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "kwsys",
+  "version-date": "2021-08-06",
+  "description": "Provides platform-independent APIs to many common system features",
+  "homepage": "https://gitlab.kitware.com/utils/kwsys",
+  "supports": "!uwp",
+  "dependencise": [
```
Comment (JonLiu1993) [inline ports/kwsys/vcpkg.json:7]:
```suggestion
  "dependencies": [
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/kwsys/portfile.cmake [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    OPTIONS_DEBUG
+		-DKWSYS_INSTALL_BIN_DIR=${CURRENT_PACKAGES_DIR}/debug/bin
+    -DKWSYS_INSTALL_LIB_DIR=${CURRENT_PACKAGES_DIR}/debug/lib
+)
+
+vcpkg_cmake_install()
+
+
+# Handle copyright
+file(INSTALL ${SOURCE_PATH}/Copyright.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (JonLiu1993) [inline ports/kwsys/portfile.cmake:28]:
```suggestion
file(INSTALL "${SOURCE_PATH}/Copyright.txt" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Review by JonLiu1993 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
LGTM, thanks for the new library!

**Pull request merged**